File: backend/app/models/recipe_import_job.py
"""Trwały stan importu przepisu przez AI, niezależny od połączenia telefonu."""

from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text, func
from sqlalchemy.dialects.postgresql import JSON, UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base


class RecipeImportJob(Base):
    __tablename__ = "recipe_import_jobs"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="queued", index=True)
    # Dane wejściowe usuwamy po zakończeniu — zwłaszcza zdjęcia nie powinny
    # pozostawać w historii zadań obok gotowego przepisu.
    payload: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    recipe_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True), ForeignKey("recipes.id", ondelete="SET NULL"), nullable=True
    )
    error: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    started_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    heartbeat_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    finished_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

File: frontend/lib/models/recipe_import_job.dart
class RecipeImportJob {
  final String id;
  final String status;
  final String? recipeId;
  final String? error;

  const RecipeImportJob({
    required this.id,
    required this.status,
    this.recipeId,
    this.error,
  });

  bool get isActive => status == 'queued' || status == 'processing';

  factory RecipeImportJob.fromJson(Map<String, dynamic> json) => RecipeImportJob(
        id: json['id'] as String,
        status: json['status'] as String,
        recipeId: json['recipe_id'] as String?,
        error: json['error'] as String?,
      );
}
